src/builder.rs

```diff
@@ -337,3 +325,42 @@ impl AstarteBuilder {
         Ok(device)
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::AstarteBuilder;
+
+    #[test]
+    fn builder() {
+        let realm = "test";
+        let device_id = "xxxxxxxxxxxxxxxxxxxxxxx";
+        let credentials_secret = "xxxxxxxxxxxxxxxxx/xxxxxxxxxxxxxxxxxxxxxxxxxx";
+        let pairing_url = "https://api.example.com/pairing";
+
+        let mut sdk_builder =
```

Comment:
Have you by any chance considered the possibility of libraries such as [this one](https://github.com/idanarye/rust-typed-builder) for a builder?

Reply:
I have considered it and decided against it, it's not worth to have another dependency for this relatively simple task, and what that library does is not a great fit for this project